tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs

```diff
@@ -999,19 +999,43 @@ public string IgnoreErrorsAndIgnoreClassesCombineTests(string[] ignoreClasses, s
             return string.Join(",", _defaultConfig.IgnoreErrorsConfiguration.Keys);
         }
 
-        [TestCase("401", new[] { "405" }, ExpectedResult = new[] { "405" })]
-        [TestCase("401", new string[0], ExpectedResult = new string[0])]
-        [TestCase("401", null, ExpectedResult = new[] { "401" })]
-        public string[] ExpectedStatusCodesSetFromLocalAndServerOverrides(string local, string[] server)
+        [TestCase("401", new[] { "405" }, null,       ExpectedResult = new[] { "405" })]
+        [TestCase("401", new string[0],   null,       ExpectedResult = new string[0])]
+        [TestCase("401", null,            null,       ExpectedResult = new[] { "401" })]
+        [TestCase(null,  null,            "401",      ExpectedResult = new[] { "401" })]
+        [TestCase(null,  new[] { "405" }, "401",      ExpectedResult = new[] { "401" })]
+        [TestCase("402", new string[0],   "401",      ExpectedResult = new[] { "401" })]
+        [TestCase("402", new string[0],   "401, 503", ExpectedResult = new[] { "401", "503" })]
+        public string[] ExpectedStatusCodesSetFromLocalServerAndEnvironmentOverrides(string local, string[] server, string env)
```

Comment:
Should there be tests for ranges in the environment variable?

Reply:
There could be, but in this testcase it would only validate that any given string with a hyphen in it isn't mangled by the EnvironmentOverrides method.  We're asserting on the value of `ExpectedErrorStatusCodesForAgentSettings` which is just the same string array returned by the `EnvironmentOverrides(ServerOverrides(.....))` call.

We'd have to do something more like the logic in the `ExpectedStatusCodesParserTests` test case right below these in order to assert on the actual `ExpectedStatusCodes` MatchRule values.

Reply:
I just want to be sure that we have a test case that covers that we can handle status code ranges in the environment variable. I'll defer finding an appropriate testing point to you.

Reply:
Put in a testcase with a range string, for what it's worth.